### DISCUSSION TITLE
[client-preset] I want to add `/* eslint-disable */` to fragment-masking.ts

### DISCUSSION THREAD

**Discussion by MH4GF** [dotansimha/graphql-code-generator#8819] [Ideas] [answered]
Hello! I am a fan of graphql-code-generator! I am now trying to implement client-preset.
Of the code generated by this preset, only fragment-masking.ts is not eslint-disable. Is there any reason why you have not done so?

Also, while it seems easy enough to add, I am wondering if this change should be marked as a breaking change.

Thanks!

**Comment by saihaj** [comment, marked as answer]
usually you would want to avoid linting generated files. But if you are running then codegen plugin system allows you to simply add yourself without codegen being opinionated on what to disable and what not to.

You can use [@graphql-codegen/add](https://the-guild.dev/graphql/codegen/plugins/other/add#examples).

Duplicate: https://github.com/dotansimha/graphql-code-generator/issues/8807

**Reply by MH4GF** [reply]
Thank you for response! I tried to use the add plugin too, but my understanding is that it is not possible to pass plugins from codegen.ts to fragment-masking.ts.

I could pass plugins to graphql.ts, which is generated at the same time by client-preset, 
https://github.com/dotansimha/graphql-code-generator/blob/24d91bb425414082520bacbf7d97bc59d1c62b43/packages/presets/client/src/index.ts?plain=1#L132

but I could not seem to pass other plugins to fragment-masking.ts.
https://github.com/dotansimha/graphql-code-generator/blob/24d91bb425414082520bacbf7d97bc59d1c62b43/packages/presets/client/src/index.ts?plain=1#L162

**Reply by eperez-zuora** [reply]
@MH4GF did you found a solution?

**Reply by MH4GF** [reply]
@eperez-zuora I solved this problem by adding directly to .eslintignore .
https://eslint.org/docs/latest/use/configure/ignore

**Reply by espipj** [reply]
I would say this is a bug could it be further investigated?

**Reply by Sashkan** [reply]
Any update on this ? I'm not fan of updating the `.eslintignore` to handle something that only affects a fragment (huhu) of the generated codebase

**Reply by hyunchulkwak** [reply]
I added this to codegen.ts to override fragment-masking config. Thanks to @MH4GF 

```ts
import { preset } from '@graphql-codegen/client-preset';
import * as addPlugin from '@graphql-codegen/add';

const originalBuildGeneratesSection = preset.buildGeneratesSection;
preset.buildGeneratesSection = (...args) => {
  const result = originalBuildGeneratesSection(...args);

  if (!Array.isArray(result)) return result;

  result.forEach(({ pluginMap, plugins }) => {
    if ('fragment-masking' in pluginMap) {
      pluginMap['add'] = addPlugin;
      plugins.push({ add: { content: '/* eslint-disable */' } });
    }
  })

  return result;
};
```

**Reply by MH4GF** [reply]
It would be nice to solve this on the client-preset side if possible, but this is a great work-around. Thanks! @hyunchulkwak
